### DISCUSSION TITLE
Return error as string in C

### DISCUSSION THREAD

**Discussion by ghost** [community/community#125205] [Programming Help] [answered]
#include <stdio.h>

char* def(int num)
{
    if(num<10&&num!=1)
    {
        char result[2] = {num+48,'\0'};
        return result;
    }
}
int main()
{
    printf("%s", def(6));
}

In this code, 6 should be output, but it outputs (null)
Is there a solution?

**Comment by davevad93** [comment, marked as answer]
Hi @typetypetype2 ,

I'm not a `C` expert but i think it's because of how the `def` function is returning the result, try with this it should work:

```c
#include <stdio.h>

char* def(int num)
{
    static char result[2];
    if(num<10 && num!=1)
    {
        result[0] = num+48;
        result[1] = '\0';
        return result;
    }
    return NULL;
}

int main()
{
    printf("%s\n", def(6));
    return 0;
}
```
